src/common/config/store/store.go

```diff
@@ -0,0 +1,116 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"github.com/goharbor/harbor/src/common/config/metadata"
+	"github.com/goharbor/harbor/src/common/config/store/driver"
+	"github.com/goharbor/harbor/src/common/http/modifier/auth"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"sync"
+)
+
+// Driver ...
+type Driver interface {
+	// Load - load config item from config driver
+	Load() (map[string]interface{}, error)
+	// Save - save config item into config driver
+	Save(cfg map[string]interface{}) error
+}
+
+// ConfigStore - the config data store
+type ConfigStore struct {
```

Comment:
Let's explicitly document/comment that the `Store` datastructure is not for external usage, user should only use `cfgmangager`